packages/react-router/lib/server-runtime/routeModules.ts

```diff
@@ -1,95 +1,15 @@
-import type { Location } from "../router/history";
+import type { ActionFunction, LoaderFunction } from "../router/utils";
 import type {
-  ActionFunction as RRActionFunction,
-  ActionFunctionArgs as RRActionFunctionArgs,
-  AgnosticRouteMatch,
-  LoaderFunction as RRLoaderFunction,
-  LoaderFunctionArgs as RRLoaderFunctionArgs,
-  Params,
-} from "../router/utils";
-import type { AppData, AppLoadContext } from "./data";
-import type { SerializeFrom } from "../dom/ssr/components";
-import type { LinkDescriptor } from "../router/links";
+  ClientActionFunction,
+  ClientLoaderFunction,
+  LinksFunction,
+  MetaFunction,
+} from "../dom/ssr/routeModules";
 
 export interface RouteModules<RouteModule> {
   [routeId: string]: RouteModule | undefined;
 }
 
-/**
- * @deprecated Use `LoaderFunctionArgs`/`ActionFunctionArgs` instead
- */
-export type DataFunctionArgs = RRActionFunctionArgs<AppLoadContext> &
-  RRLoaderFunctionArgs<AppLoadContext> & {
-    // Context is always provided in Remix, and typed for module augmentation support.
-    // RR also doesn't export DataFunctionArgs, so we extend the two interfaces here
-    // even tough they're identical under the hood
-    context: AppLoadContext;
-  };
-
-/**
- * A function that handles data mutations for a route on the server
- */
-export type ActionFunction = (
-  args: ActionFunctionArgs
-) => ReturnType<RRActionFunction>;
-
-/**
- * Arguments passed to a route `action` function
- */
-export type ActionFunctionArgs = RRActionFunctionArgs<AppLoadContext> & {
-  // Context is always provided in Remix, and typed for module augmentation support.
-  context: AppLoadContext;
-};
-
-/**
- * A function that handles data mutations for a route on the client
- * @private Public API is exported from @react-router/react
- */
-type ClientActionFunction = (
```

Comment:
These client types were duped here and in the `@remix-run/react` package so we just keep the other ones.